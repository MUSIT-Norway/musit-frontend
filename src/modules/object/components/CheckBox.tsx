import * as React from 'react';

export const CheckBox = (props: {
  id: string;
  checked: boolean;
  displayValue: string;
  onChange: Function;
  viewMode?: boolean;
}) => (
  <div>
    <label>
      <input
        className="checkBoxComponent"
        type="checkbox"
        id={props.id}
        checked={props.checked}
        disabled={props.viewMode}
        onChange={() => props.onChange()}
      />
      {props.displayValue}
    </label>
  </div>
);
