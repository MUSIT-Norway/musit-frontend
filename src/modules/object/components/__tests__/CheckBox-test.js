import React from 'react';
import { shallow } from 'enzyme';
import { CheckBox } from '../CheckBox';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

describe('Test for checkBox component.', () => {
  it('Snapshot test for checkBox component', () => {
    const checkbox = shallow(
      <CheckBox id="TestId" checked={true} displayValue="Hello" onChange={e => e} />
    );
    expect(shallowToJson(checkbox)).toMatchSnapshot();
  });

  it('CheckBox onChange event', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(
      <CheckBox id="TestId2" checked={false} displayValue="Hello" onChange={onChange} />
    );
    const comp = wrapper.find('.checkBoxComponent');
    comp.simulate('change');

    expect(onChange.calledOnce).toBe(true);
    expect(comp.node.props.id).toBe('TestId2');
  });

  it('Render CheckBox text', () => {
    const onChange = sinon.spy();
    const wrapper = shallow(
      <CheckBox id="TestId2" checked={false} displayValue="Hello" onChange={onChange} />
    );
    expect(wrapper.text()).toBe('Hello');
  });
});
