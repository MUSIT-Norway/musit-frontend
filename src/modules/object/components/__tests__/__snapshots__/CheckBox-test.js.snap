// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for checkBox component. Snapshot test for checkBox component 1`] = `
<div>
  <label>
    <input
      checked={true}
      className="checkBoxComponent"
      id="TestId"
      onChange={[Function]}
      type="checkbox"
    />
    Hello
  </label>
</div>
`;
